src/main/java/com/agricraft/agricore/plant/AgriPlant.java

```diff
@@ -33,6 +38,36 @@
     private final AgriRequirement requirement;
     private final AgriTexture texture;
 
+    public AgriPlant(AgriSerializable as) {
+        final AgriPlantOld old = (AgriPlantOld) as;
```

Comment:
This cast pains me. Shouldn't the input parameter be constrained to AgriPlantOld?

Reply:
Not completely certain, referring to your earlier point about versioning the jsons.